fairmq/tools/MakeUnique.h

```diff
@@ -0,0 +1,41 @@
+/********************************************************************************
+ *    Copyright (C) 2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#ifndef FAIR_MQ_TOOLS_MAKEUNIQUE_H
+#define FAIR_MQ_TOOLS_MAKEUNIQUE_H
+
+#include <cstddef>           // for std::size_t
+#include <memory>            // for std::unique_ptr
+#include <memory_resource>   // for std::pmr::memory_resource
+#include <type_traits>       // for std::forward
+
+namespace fair::mq::tools {
+
+// similar to std::make_unique, but with support for
+// * allocator (pmr only! as defined in
+//   https://en.cppreference.com/w/cpp/memory/memory_resource),
+// * alignment, and
+// * right padding (allocating a larger heap buffer than strictly needed for the
+//   given T)
+template<typename T, typename... Args, typename Allocator = std::pmr::memory_resource>
+constexpr auto MakePaddedUnique(std::size_t size_hint, /* = sizeof(T) */
+                                std::size_t alignment, /* = alignof(std::max_align_t) */
+                                Allocator* pmr,        /* = std::pmr::get_default_resource() */
+                                Args&&... args)
+{
+    auto const size = (size_hint > sizeof(T)) ? size_hint : sizeof(T);
```

Comment:
Would it be more clear to call this entire option `shm-metadata-padding`?
1. to me it is more clear from the name what it does. With the current name, having default size at 0 is kind of strange, although you do explain the condition in the docs.
2. you don't need the if/else here, just size+padding.

Reply:
I see your point, however, I kept the command line option as is (`--shm-metadata-msg-size`), because from there the user does not know anything about the actual size of the metadata msg, so it would be strange to just tune the padding. But in the C++ interfaces I changed it to padding, because it simplifies the code.